packages/yourloops/components/device/utils/device.utils.ts

```diff
@@ -110,20 +111,27 @@ export const formatParameterValue = (value: string | number, units: string | Uni
   return value.toFixed(numberOfDecimals)
 }
 
-export const sortHistoryParametersByDate = (historyParameters: ParametersChange[]): ParametersChange[] => {
-  return historyParameters.sort((a, b) => {
-    return new Date(a.changeDate).valueOf() - new Date(b.changeDate).valueOf()
+export const sortHistoryParametersByDate = (historyParameters: ParametersChange[]): void => {
+  historyParameters.sort((a, b) => {
+    return new Date(b.changeDate).valueOf() - new Date(a.changeDate).valueOf()
   })
 }
 
-export const sortPumpSettingsParameterByLevel = (historyParameters: ParametersChange[]): ParametersChange[] => {
+export const sortPumpSettingsParametersByDate = (historyParameters: ParametersChange[]): void => {
+  historyParameters.forEach((parameterChange) => {
+    parameterChange.parameters.sort((paramA, paramB) => {
+      return sortByDate(paramB.effectiveDate, paramA.effectiveDate)
+    })
+  })
+}
+
+export const sortPumpSettingsParameterByLevel = (historyParameters: ParametersChange[]): void => {
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏`Parameters` instead of `Parameter` in the method name

```suggestion
export const sortPumpSettingsParametersByLevel = (historyParameters: ParametersChange[]): void => {
```